pkg/sql/plan2/function/builtin/unary/lengthutf8.go

```diff
@@ -23,29 +24,36 @@ import (
 	"github.com/matrixorigin/matrixone/pkg/vm/process"
 )
 
+var (
+	errorParameterIsNotString = errors.New("the parameter is not char or varchar")
+	errorParameterIsInvalid   = errors.New("invalid parameter")
+)
+
 func LengthUTF8(vectors []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	if len(vectors) == 0 || proc == nil {
+		return nil, errorParameterIsInvalid
+	}
+	if vectors[0] == nil {
+		return nil, errorParameterIsInvalid
+	}
 	inputVector := vectors[0]
 	resultType := types.Type{Oid: types.T_uint64, Size: 8}
 	resultElementSize := int(resultType.Size)
-	if inputVector.IsScalar() {
-		if inputVector.ConstVectorIsNull() {
-			return proc.AllocScalarNullVector(resultType), nil
-		}
-		inputValues := inputVector.Col.(*types.Bytes)
-		resultVector := vector.NewConst(resultType)
-		resultValues := make([]uint64, 1)
-		vector.SetCol(resultVector, lengthutf8.StrLengthUTF8(inputValues, resultValues))
-		return resultVector, nil
-	} else {
-		inputValues := inputVector.Col.(*types.Bytes)
-		resultVector, err := proc.AllocVector(resultType, int64(resultElementSize*len(inputValues.Lengths)))
-		if err != nil {
-			return nil, err
-		}
-		resultValues := encoding.DecodeUint64Slice(resultVector.Data)
-		resultValues = resultValues[:len(inputValues.Lengths)]
-		nulls.Set(resultVector.Nsp, inputVector.Nsp)
-		vector.SetCol(resultVector, lengthutf8.StrLengthUTF8(inputValues, resultValues))
-		return resultVector, nil
+	if inputVector.IsScalar() && inputVector.ConstVectorIsNull() {
+		return proc.AllocScalarNullVector(resultType), nil
+	}
+
+	inputValues, ok := inputVector.Col.(*types.Bytes)
```

Comment:
so when our input vector is a const(not null), you don't wanna output a const vector?

this may cause problems in queries like this:
```sql
select a, length("hello") from table;
```

Reply:
I will not cause problems.

I may understand the logic before mine. 
When the `inputVector` looks like  ["abc","abc","abc","abc","abc",],  it evaluates the result just once.
That sounds good.
I will undo the change.